packages/ckeditor5-dev-env/tests/release-tools/utils/generatechangelogfromcommits-integration.js

```diff
@@ -121,6 +121,65 @@ describe( 'dev-env/release-tools/utils', () => {
 					release( '0.2.1' );
 				} );
 		} );
+
+		it( 'does not hoist issues from the commit body with additional notes for merge commit', () => {
+			exec( 'git commit --allow-empty ' +
+				'--message "Merge pull request #7 from ckeditor/t/6" ' +
+				'--message "Other: Some docs improvements. Closes #6." ' +
+				'--message "Did you see the #3 and #4?" ' +
+				'--message "NOTE: Please read #1." ' +
+				'--message "BREAKING CHANGES: Some breaking change." ' );
+
+			return generateChangelog( '0.3.0', '0.2.1' )
+				.then( () => {
+					const latestChangelog = replaceCommitIds( getChangesForVersion( '0.3.0' ) );
+
+					expect( latestChangelog.split( '\n' ).length ).to.equal( 13 );
+
+					const changesAsArray = latestChangelog.split( '\n' ).filter( ( line ) => line.trim().length );
+
+					//jscs:disable maximumLineLength
+					expect( changesAsArray[ 0 ] ).to.equal( '### Other changes' );
```

Comment:
It's again more complex than it needs to be. Instead of checking the number of new lines and then filtering them out and then checking every line separately you could compare it using one big multiline template string. It'd be closest to how you'd check it manually (by looking into the changelog).